pkg/oidc/oidc.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oidc
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/hcl/v2"
+)
+
+const (
+	defaultClientID      = "gangway"
+	defaultUsernameClaim = "email"
+	defaultGroupsClaim   = "groups"
+)
+
+// Config deals with providing OIDC related fields to the Lokomotive
+// cluster for OIDC based Authentication via Dex and Gangway.
+type Config struct {
+	IssuerURL     string `hcl:"issuer_url,optional"`
+	ClientID      string `hcl:"client_id,optional"`
+	UsernameClaim string `hcl:"username_claim,optional"`
+	GroupsClaim   string `hcl:"groups_claim,optional"`
+}
+
+// newConfig returns a new config with defaults values.
+func newConfig() *Config {
+	return &Config{
+		IssuerURL:     "",
```

Comment:
This line can be removed, so we make sure of Go zero-value of string, which is exactly `""`.
```suggestion

```